packages/xstate-react/test/useMachine.test.tsx

```diff
@@ -987,3 +988,113 @@ describe('useMachine (strict mode)', () => {
     expect(container.textContent).toBe('2');
   });
 });
+
+describe('input', () => {
+  it('Should provide the input on the state', async () => {
+    const model = createModel({}).withInput(
+      {} as {
+        foo: string;
+      }
+    );
+
+    const machine = model.createMachine({}, {});
+
+    const App = () => {
+      const [state] = useMachine(machine, {
+        input: {
+          foo: 'bar'
+        }
+      });
+      return <>{state.input.foo}</>;
+    };
+
+    const { container } = render(<App />);
+
+    expect(container.textContent).toBe('bar');
+  });
+
+  it('Should keep the input up to date', () => {
+    const model = createModel({}).withInput(
+      {} as {
+        foo: string;
+      }
+    );
+
+    const machine = model.createMachine({}, {});
+
+    const App = (props: { foo: string }) => {
+      const [state] = useMachine(machine, {
+        input: {
+          foo: props.foo
+        }
+      });
+      return <>{state.input.foo}</>;
```

Comment:
a question: is it expected for the user that sometimes this isn't true: `props.foo === state.input.foo`?